### PR TITLE
Integrate api into myCommunityStake page

### PR DESCRIPTION
**Pull request by kurtisassad**
## Link to Issue
Closes: #7238

## Description of Changes
- Makes changes to GetStakeTransactions to pull the correct data for the front end
- Replaces the hardcoded sample data with the data returned from the API
- Modifies buildEtherscanLink to link to either blast or base depending on the chain node of the community

## Test Plan
Workflow:
- Create a community with stake and start buying/selling stake
- On transaction buy/sell check the etherscan page (pop-out icon). Make sure it navigates to the correct etherscan.
- Check this transaction page.

Manual:
- You can enter your own transactions into the database, just replace your address. For example:
```
insert into "StakeTransactions" values ('0x1234567891234567891234567891234567891234567891234567891234567891', 'zaks-community', 2, [Your address], 100, 1000, 'sell', 100);
insert into "StakeTransactions" values ('0x1234567891234567891234567891234567891234567891234567891234567893', 'jpegmullets', 2, [Your address], 100, 1000, 'buy', 100);
insert into "StakeTransactions" values ('0x1234567891234567891234567891234567891234567891224567891234567893', 'blast-off', 2, [Your address], 100, 1000, 'buy', 100);
```

- check that blast-off links to the blastscan page, and jpegmullets links to the basescan page by clicking on the right hand side of the transaction listing (window pop-out icon)

Looks like:
image
image

## Deployment Plan
1. (Backfill transactions) We will need to go through base and blast and manually enter the transactions that were made before this PR had gone in

### CODE REVIEW

File: packages/commonwealth/client/scripts/hooks/useTransactionHistory.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+const useTransactionHistory = ({
+  filterOptions,
+  addressFilter,
+}: TransactionHistoryProps) => {
+  const { data } = trpc.community.getStakeTransaction.useQuery({
+    addresses: addressFilter.length > 0 ? addressFilter.join(',') : undefined,
+  });
+
+  const WEI_PER_ETHER = 1000000000000000000;
```
Comment (Rotorsoft) [inline packages/commonwealth/client/scripts/hooks/useTransactionHistory.ts:18]: should this be a global constant in procotol domain or just used here?

Comment (kurtisassad) [reply packages/commonwealth/client/scripts/hooks/useTransactionHistory.ts:18]: Yep it is global, I will find some place shared file to put it

File: libs/model/src/community/GetStakeTransaction.query.ts [hunk 2]
```diff
@@ -8,27 +9,35 @@ export const GetStakeTransaction: Query<
...
+         json_build_object(
+           'id', c.id,
+           'default_symbol', c.default_symbol,
+           'icon_url', c.icon_url,
+           'name', c.name
+         ) AS community
+       FROM "StakeTransactions" AS t
+       LEFT JOIN "Communities" AS c ON c.id = t.community_id
+       LEFT JOIN "CommunityStakes" AS cs ON cs.community_id = t.community_id
+       WHERE :addresses IS NULL OR t.address IN (:addresses);
```
[the remark this answers is not in the document]
Comment (kurtisassad) [reply libs/model/src/community/GetStakeTransaction.query.ts:33]: Currently we don't have a large number of transactions, also the front end has not been set up to perform pagination, so maybe we can add pagination in the future.

File: packages/commonwealth/client/scripts/hooks/useTransactionHistory.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        action: t.stake_direction === 'buy' ? 'mint' : 'burn',
+        totalPrice: `${(parseFloat(t.stake_price) / WEI_PER_ETHER).toFixed(
+          5,
+        )} ETH`,
+        avgPrice: `${(
+          parseFloat(t.stake_price) /
+          WEI_PER_ETHER /
+          t.stake_amount
+        ).toFixed(5)} ETH`,
+        etherscanLink: buildEtherscanLink(t.transaction_hash),
```
[the remark this answers is not in the document]
Comment (kurtisassad) [reply packages/commonwealth/client/scripts/hooks/useTransactionHistory.ts]: Good catch, fixed and updated test plan

**Review by jnaviask — CHANGES_REQUESTED** [review]
One last comment then g2g.

**Comment by kurtisassad** [comment]
Drafting because I broke transaction buying etherscan link

**Comment by kurtisassad** [comment]
Fixed and added to test plan

**Pull request merged**